F# build is broken due to new String.StartsWith overload

## Steps to Reproduce

1. Build FSharp
2. It fails during build

This was most likely introduced because https://github.com/mono/mono/pull/7492 introduced new overloads on String from corefx.

## Current Behavior

```
"@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj" (default target) (1) ->
(CopyAndSubstituteTextFiles target) -> 
  @/external/bockbuild/builds/fsharp-fsharp-561af8b/src/scripts/scriptlib.fsx(40,39): error FS0041: A unique overload for method 'StartsWith' could not be determined based on type information prior to this program point. A type annotation may be needed. Candidates: String.StartsWith(value: char) : bool, String.StartsWith(value: string) : bool [@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj]
  @/external/bockbuild/builds/fsharp-fsharp-561af8b/src/FSharpSource.targets(207,5): error MSB3073: The command "mono @/external/bockbuild/builds/fsharp-fsharp-561af8b/packages/FSharp.Compiler.Tools.4.1.27/tools/fsi.exe --exec "@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/scripts/subst.fsx" --in:"@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build/Microsoft.FSharp.Targets" --out:"@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build-proto/../../../Proto/net40/bin/Microsoft.FSharp-proto.targets" --pattern1:"{BuildSuffix}" --replacement1:"-proto" --pattern2:"" --replacement2:"" " exited with code 1. [@/external/bockbuild/builds/fsharp-fsharp-561af8b/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj]

    0 Warning(s)
    2 Error(s)
```

## Expected Behavior

No build failure.

### On which platforms did you notice this

[ X ] macOS
[ ] Linux
[ ] Windows

**Version Used**: master

--- a/packaging/MacSDK/fsharp.py
+++ b/packaging/MacSDK/fsharp.py
@@ -8,7 +8,7 @@ class FsharpPackage(GitHubTarballPackage):
             override_properties={ 'make': 'make' })
 
         self.extra_stage_files = ['lib/mono/xbuild/Microsoft/VisualStudio/v/FSharp/Microsoft.FSharp.Targets']
-        self.sources.extend(['patches/fsharp-portable-pdb.patch'])
+        self.sources.extend(['patches/fsharp-portable-pdb.patch', 'patches/fsharp-string-switchName.patch'])
 
     def prep(self):
         Package.prep(self)

--- a/packaging/MacSDK/patches/fsharp-string-switchName.patch
+++ b/packaging/MacSDK/patches/fsharp-string-switchName.patch
@@ -1,0 +1,13 @@
+diff --git a/src/scripts/scriptlib.fsx b/src/scripts/scriptlib.fsx
+index cc797e305..5a7be7d2b 100644
+--- a/src/scripts/scriptlib.fsx
++++ b/src/scripts/scriptlib.fsx
+@@ -36,7 +36,7 @@ module Scripting =
+ #if INTERACTIVE
+     let argv = Microsoft.FSharp.Compiler.Interactive.Settings.fsi.CommandLineArgs |> Seq.skip 1 |> Seq.toArray
+ 
+-    let getCmdLineArgOptional switchName = 
++    let getCmdLineArgOptional (switchName: string) =
+         argv |> Array.filter(fun t -> t.StartsWith(switchName)) |> Array.map(fun t -> t.Remove(0, switchName.Length).Trim()) |> Array.tryHead 
+ 
+     let getCmdLineArg switchName defaultValue = 

